moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1262,6 +1263,26 @@ bool PlanningScene::usePlanningSceneMsg(const moveit_msgs::PlanningScene& scene_
     return setPlanningSceneMsg(scene_msg);
 }
 
+collision_detection::OccMapTreePtr createOctomap(const octomap_msgs::Octomap& map)
+{
+  std::shared_ptr<collision_detection::OccMapTree> om =
+      std::make_shared<collision_detection::OccMapTree>(map.resolution);
+  if (map.binary)
+  {
+    octomap_msgs::readTree(om.get(), map);
+  }
+  else
+  {
+    std::stringstream datastream;
+    if (map.data.size() > 0)
+    {
+      datastream.write((const char*)&map.data[0], map.data.size());
+      om->readData(datastream);
+    }
```

Comment:
@JStech I don't know enough about octomaps to know what this is doing.  Do you have any feedback for this?

Reply:
@simonschmeisser copied the logic from [`octomap_msgs`](https://github.com/OctoMap/octomap_msgs/blob/melodic-devel/include/octomap_msgs/conversions.h#L110). We can't use the upstream methods because they [instantiate a `octomap::OcTree` directly](https://github.com/OctoMap/octomap_msgs/blob/melodic-devel/include/octomap_msgs/conversions.h#L95) and we want the lockable wrapper `OccMapTree` instead.

Reply:
Yeah, I'm no expert on Octomap internals, but this looks like it's just copying the serialized tree from the message into the `om` object.

Reply:
It is possible to exit out of this function without having any useful data in the OccMapTree. The octomap message could have empty data array. Would it make sense to have a warning here if the data is empty and we end up adding basically nothing to the collision scene?

Reply:
Good point. Both methods that use this helper have a guard for `!msg.data.empty()`, so I believe the case can't happen at the moment. Pondering this for a while I would prefer to allow the use-case though. `octomap_msgs::readTree` explicitly allows for this and this method is basically just a wrapper for it.

Reply:
I cannot follow, why shouldn't an octomap potentially be empty?

Reply:
I think this helper function is a great candidate for adding a unit test for.

Reply:
Please address this in a follow-on PR if you get a chance.  I merged this PR as it's a useful fix and has been sitting long enough.